amd_openvx_extensions/amd_rpp/source/tensor/ToDecibels.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+struct ToDecibelsLocalData {
+    vxRppHandle *handle;
+    Rpp32u deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pDst;
+    Rpp32f cutOffDB;
+    Rpp32f multiplier;
+    Rpp32f referenceMagnitude;
+    RpptDescPtr pSrcDesc;
+    RpptDescPtr pDstDesc;
+    RpptImagePatch *pSrcDims;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+void copy_src_dims_and_update_dst_roi(ToDecibelsLocalData *data, RpptROI *src_roi, RpptROI *dst_roi) {
+    memcpy(dst_roi, src_roi, data->pSrcDesc->n * sizeof(RpptROI));
+    for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
+        data->pSrcDims[i].width = src_roi[i].xywhROI.roiWidth;
+        data->pSrcDims[i].height = src_roi[i].xywhROI.roiHeight;
+    }
```

Comment:
```suggestion
    for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
        data->pSrcDims[i].width = src_roi[i].xywhROI.roiWidth;
        data->pSrcDims[i].height = src_roi[i].xywhROI.roiHeight;
    }
```

This part can happen in refresh itself right?